Realm.Shared/RealmList.cs

```diff
@@ -32,14 +32,14 @@ namespace Realms
     /// </remarks>
     /// 
     /// <typeparam name="T">Type of the RealmObject which is the target of the relationship.</typeparam>
-    public class RealmList<T> : IList<T> where T : RealmObject
+    public class RealmList<T> : IList<T>, ICopyValuesFrom where T : RealmObject
     {
         private class RealmListEnumerator : IEnumerator<T> 
         {
             private int index;
-            private RealmList<T> enumerating;
+            private IList<T> enumerating;
```

Comment:
We don't *require* it to be a `RealmList` though, but yeah it could be reverted and avoids dispatch through the interface.